pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -171,10 +172,21 @@ func (rc *reconciler) reconcile() {
 	// Ensure volumes that should be unmounted are unmounted.
 	for _, mountedVolume := range rc.actualStateOfWorld.GetMountedVolumes() {
 		if !rc.desiredStateOfWorld.PodExistsInVolume(mountedVolume.PodName, mountedVolume.VolumeName) {
-			// Volume is mounted, unmount it
-			glog.V(12).Infof(mountedVolume.GenerateMsgDetailed("Starting operationExecutor.UnmountVolume", ""))
-			err := rc.operationExecutor.UnmountVolume(
-				mountedVolume.MountedVolume, rc.actualStateOfWorld)
+			var err error
+			pvVolumeMode := volumehelper.GetVolumeMode(mountedVolume.MountedVolume.VolumeSpec)
+			if pvVolumeMode != v1.PersistentVolumeBlock {
```

Comment:
Instead of if/else everywhere, can we make an interface and have a fs implementation and a block implementation?